documents/sd/chapter/Classes/Common/Events/Events.tex

```diff
@@ -11,101 +11,314 @@ \subsection*{MORR.Shared.Events}
+	
+	\begin{methods}
+		\begin{method}{public void Enqueue(object @event)}{Asynchronously enqueues a new event of type T.}
             \begin{parameters}
                 \para{T @event}{The event to enqueue.}
             \end{parameters}
+            \begin{exceptions}
+            	\excp{ArgumentException}{Exception is thrown when event type is not T}
+            \end{exceptions}
         \end{method}
     \end{methods}
 \end{absclass}
+	
+\subsection*{MORR.Shared.Events.Queue.Strategy}
 
+\begin{class}{ChannelConsumingException}
+	\clsdiagram[width=0.4\textwidth]{resources/Classes/Common/Shared/Events/Queue/Strategy/ChannelConsumingException.png}
+	
+	\clsdcl{public class ChannelConsumingException : Exception}	
+	
+	\clsdsp{A simple Exception that encapsulates errors occuring in ChannelStrategies.}
+	
+\end{class}
+	
 \begin{interface}{IEventQueueStorageStrategy}
-    \clsdiagram[width=0.4\textwidth]{resources/Classes/Common/Shared/Events/IEventQueueStorageStrategy.png}
+    \clsdiagram[width=0.5\textwidth]{resources/Classes/Common/Shared/Events/Queue/Strategy/IEventQueueStorageStrategy.png}
 
     \clsdcl{public interface IEventQueueStorageStrategy<T> where T : Event}
 
     \clsdsp{Provides the backing store of an EventQueue with a specific storage strategy.}
 
-    \begin{methods}
-        \begin{method}{IAsyncEnumerable<T> GetEventsInternal()}{Asynchronously gets all events as concrete type T from this storage strategy.}
-            \return{IAsyncEnumerable<T>}{A stream of events of type T.}
-        \end{method}
-        \begin{method}{void Enqueue(T @event)}{Asynchronously enqueues a new event of type T with this storage strategy.}
-            \begin{parameters}
-                \para{T @event}{The event to enqueue.}
-            \end{parameters}
-        \end{method}
-    \end{methods}
 \end{interface}
 
-\begin{class}{RingBufferStorageStrategy}
-    \clsdiagram[width=0.4\textwidth]{resources/Classes/Common/Shared/Events/RingBufferStorageStrategy.png}
+\subsubsection*{MORR.Shared.Event.Queue.Strategy.MultiConsumer}
 
-    \clsdcl{public class RingBufferStorageStrategy<T> : IEventQueueStorageStrategy<T> where T : Event}
+\begin{class}{BoundedMultiConsumerChannelStrategy<TEvent>}
+	\clsdiagram[width=0.5\textwidth]{resources/Classes/Common/Shared/Events/Queue/Strategy/MultiConsumer/BoundedMultiConsumerChannelStrategy.png}
+	
+	\clsdcl{public class BoundedMultiConsumerChannelStrategy<TEvent> : 
+	MultiConsumerChannelStrategy<TEvent> where TEvent : Event}
+	
+	\clsdsp{A BoundedMultiConsumerChannelStrategies is a Distributive-FIFO Queue which allows multiple producers and multiple consumers. However every event is propagated to each consumer once. This queue is bounded by a buffer capacity. To be more detailed each channel is bound by the bufferCapacity. Is the capacity reached for a specific channel it will drop the oldest event and write the new event.}
```

Comment:
```suggestion
	\clsdsp{A BoundedMultiConsumerChannelStrategy is a Distributive-FIFO Queue which allows multiple producers and multiple consumers. However every event is propagated to each consumer once. This queue is bounded by a buffer capacity. To be more detailed each channel is bound by the bufferCapacity. Is the capacity reached for a specific channel it will drop the oldest event and write the new event.}
```